CalendarView.cpp

```diff
@@ -20,202 +23,281 @@ CalendarView::CalendarView(SDL_Renderer* renderer, SDL_Rect bounds, SDL_Rect gri
+		SDL_RenderDrawRect(renderer, &eventBounds);
+
+		dest.x += 50;
+		renderText((*events[dateString])[i]->eventName, false, NULL, &dest, black, defaultFont);
+		dest.x += 250;
+		renderText((*events[dateString])[i]->location, false, NULL, &dest, black, defaultFont);
+		dest.x -= 250;
+		dest.y += 16;
+		renderText(startTime, false, NULL, &dest, grey, defaultFont);
+		dest.x += 72;
+		renderText("-", false, NULL, &dest, grey, defaultFont);
+		dest.x += 12;
+		renderText(endTime, false, NULL, &dest, grey, defaultFont);
+
+		dest = { eventBounds.x + eventBounds.w - 23, eventBounds.y + 3, 20, 20 };
+		SDL_RenderDrawRect(renderer, &dest);
+		dest.x += 10;
+		dest.y += 10;
+		renderText("X", true, NULL, &dest, black, defaultFont);
+		dest.y += 14;
+		dest.x -= 10;
+		SDL_RenderDrawRect(renderer, &dest);
+		dest.x += 10;
+		dest.y += 10;
+		renderText("E", true, NULL, &dest, black, defaultFont);
+
+		eventBounds.y += 53;
 	}
+
+	SDL_SetRenderDrawColor(renderer, 0, 0, 0, 255);
 }
 
-void CalendarView::labelDay(std::vector<CalendarEvent*> events)
+void CalendarView::drawAddEvent()
 {
 	SDL_Color black = { 0, 0, 0, 255 };
 	SDL_Color grey = { 180, 180, 180, 255 };
-	SDL_Rect dest = { bounds.x + bounds.w / 2, bounds.y + (gridBounds.y - bounds.y) / 2, 0, 0 };
-	renderText(monthNames[Time::cMonth] + " " + Time::toOrdinal(Time::cDay) + ", " + std::to_string(Time::cYear), true, dest, black);
+	SDL_Rect dest = { bounds.x + bounds.w / 2, bounds.y + (gridBounds.y - bounds.y) / 2, 0, 0};
+	renderText("Add New Event - " + monthNames[Time::Month] + " " + Time::dayToOrdinal(Time::Day) + ", " + std::to_string(Time::Year), true, NULL, &dest, black, defaultFont);
 }
 
-void CalendarView::renderText(std::string str, bool centered, SDL_Rect dest, SDL_Color color)
+void CalendarView::drawHighlight(ViewMode currentView, SDL_Rect highlightBounds, SDL_Color highlightColor)
 {
-	SDL_Surface* temp = TTF_RenderText_Blended(defaultFont, str.c_str(), color);
-	if (centered)
+	if (highlightBounds.w == 0) { return; }
+	SDL_SetRenderDrawColor(renderer, highlightColor.r, highlightColor.g, highlightColor.b, highlightColor.a);
+	switch (currentView)
 	{
-		dest.x -= temp->w / 2;
-		dest.y -= temp->h / 2;
+	case ViewMode::Year:
+		SDL_RenderFillRect(renderer, &highlightBounds);
+		break;
+	case ViewMode::Month:
+		SDL_RenderFillRect(renderer, &highlightBounds);
+		break;
+	case ViewMode::Day:
+		break;
+	case ViewMode::AddEvents:
+		break;
 	}
 
-	dest.w = temp->w;
-	dest.h = temp->h;
-
-	SDL_Texture* text = SDL_CreateTextureFromSurface(renderer, temp);
-	SDL_FreeSurface(temp);
-	SDL_RenderCopy(renderer, text, NULL, &dest);
-	SDL_DestroyTexture(text);
+	SDL_SetRenderDrawColor(renderer, 0, 0, 0, 255);
 }
 
-void CalendarView::drawEvents(int viewMode, std::vector<CalendarEvent*> events)
+void CalendarView::drawEvents(ViewMode currentView, SDL_Rect dest, std::vector<CalendarEvent*>* events)
 {
+	SDL_Rect src = { 0, 0, 61, 0 };
+	int n = events->size() > 5 ? 5 : events->size();
```

Comment:
You're overthinking this a little.

```suggestion
	int n = std::min(events->size(), 5);
```